app/routes/questions/actions.tsx

```diff
@@ -0,0 +1,116 @@
+import {useState, useEffect, MouseEvent, useCallback} from 'react'
+import type {ActionArgs} from '@remix-run/cloudflare'
+import {Form, useSearchParams} from '@remix-run/react'
+import {redirect, json} from '@remix-run/cloudflare'
+import {likeQuestion} from '~/server-utils/stampy'
+import {DarkLight, Edit, Flag, Followup, Hide, Like, Search} from '~/components/icons-generated'
+
+export enum ActionType {
+  DARKLIGHT = 'darkLight',
+  EDIT = 'edit',
+  FLAG = 'flag',
+  FOLLOWUP = 'followup',
+  HELPFUL = 'helpful',
+  HIDE = 'hide',
+  SEARCH = 'search',
+}
+export type ActionProps = {
+  Icon: JSX.IntrinsicElements['svg']
+  title: string
+  handler?: (pageid: string) => Promise<string>
+}
+type ActionsDict = {
+  [k: string]: ActionProps
+}
+const actions = {
+  darkLight: {
+    Icon: DarkLight,
+    title: 'Dark/Light',
+  },
+  edit: {
+    Icon: Edit,
+    title: 'Edit',
+  },
+  flag: {
+    Icon: Flag,
+    title: 'Flag',
+  },
+  followup: {
+    Icon: Followup,
+    title: 'Followup',
+  },
+  helpful: {
+    Icon: Like,
+    title: 'Helpful',
+    handler: likeQuestion,
```

Comment:
this should, in theory, make adding new actions just a matter of adding an appropriate handler function to the appropriate entry here